src/S3/S3Client.php

```diff
@@ -923,19 +926,15 @@ public static function _applyRetryConfig($value, $args, HandlerList $list)
     public static function _applyApiProvider($value, array &$args, HandlerList $list)
     {
         ClientResolver::_apply_api_provider($value, $args);
-        $args['parser'] = new GetBucketLocationParser(
-            new ValidateResponseChecksumParser(
-                new AmbiguousSuccessParser(
-                    new RetryableMalformedResponseParser(
-                        $args['parser'],
-                        $args['exception_class']
-                    ),
-                    $args['error_parser'],
-                    $args['exception_class']
-                ),
-                $args['api']
-            )
+        $s3Parser = new S3Parser(
+            $args['parser'],
+            $args['error_parser'],
+            $args['api'],
+            $args['exception_class']
         );
+        $s3Parser->addS3ResultMutator('s3.get-bucket-location', new GetBucketLocationResultMutator());
```

Comment:
we can probably drop the `s3.` prefix since this is not being printed to debug logging.  The mutators are already S3-specific.